### PR TITLE
feat: fetch templates from github and extract the archive utility

### PR DESCRIPTION
**Pull request by RonenMars**
## Description
Get templates from Kittehub to have an archive, embed all of the src in a single file.
That file contains all the sources and some metadata, to be used in the UI to offer templates to users.

## Linear Ticket
https://linear.app/autokitteh/issue/UI-829/allow-to-update-templates-during-ui-run
https://linear.app/autokitteh/issue/UI-829/allow-to-update-templates-during-ui-run

## What type of PR is this? (check all applicable)

-   [x] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [ ] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/constants/templates.constants.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export const githubUrl = "https://raw.githubusercontent.com/autokitteh/kittehub/refs/heads/release/dist.zip";
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/constants/templates.constants.ts:1]: removed

File: src/locales/en/utilities/translation.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
+	"fetchAndExtract":{
+		"noFileStructure": "No file structure provided to processReadmeFiles",
+		"fetchAndExtractError": "Error in fetchAndExtract, name: {name}, error: {error}, stack: {stack}",
+		"invalidStructurePath": "Invalid structure path provided to processDirectory, path: {path}",
+		"nullOrUndefinedFound": "Null or undefined node found for {name} at {currentPath}",
+		"couldNotFindDirectory": "Could not find directory structure for {path}",
+		"skippingPathMissingMetadata": "Skipping path {path} as it is missing metadata, attributes: ",
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/locales/en/utilities/translation.json:8]: moved to the utility

File: src/utilities/fetchAndExtractZip.utils.ts [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+			});
+		} else if (isDirectoryNode(node)) {
+			files.push(...getAllFilesInDirectory(node.children, fullPath));
+		}
+	});
+
+	return files;
+};
+
+export const processReadmeFiles = (fileStructure: FileStructure | null | undefined): ProcessedRemoteCategory[] => {
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/utilities/fetchAndExtractZip.utils.ts]: refactored

File: src/utilities/fetchAndExtractZip.utils.ts [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+
+			return;
+		}
+
+		for (const [name, node] of Object.entries(structure)) {
+			if (!node) continue;
+
+			if (isDirectoryNode(node)) {
+				processDirectory(node.children, `${currentPath}/${name}`.replace(/^\//, ""));
+			} else if (isFileNode(node) && name.toLowerCase() === "readme.md") {
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/utilities/fetchAndExtractZip.utils.ts]: refactored

[the remark this answers is not in the document]
Comment (RonenMars) [reply src/utilities/fetchAndExtractZip.utils.ts]: changed to continue

**Review by J1za — APPROVED** [review]

**Pull request merged**